### ISSUE TITLE
Rainlab Blog Pagination not working as intended.

### ISSUE THREAD

**Issue by mjdwebdesign** [rainlab/blog-plugin#258]
Hello,

I'm currently using this plugin to function as a Newfeed for a website I'm developing.
The pagination and blog posts were displaying and functioning as normal before the most recent update.

I'm now having an issue with my blog posts page in that the pagination only ever directs the user to page 1, no matter which pagination option is clicked.

So, the blog posts are located at: **...url.../news/**
When any pagination option is clicked, the user is directed to: **...url.../news/1**
no matter what.

I've attempted some fixes listed by other users with a similar problem to no avail. I was hoping someone could shed some light on the situation here as I'm on a tight schedule to get this complete at current.

I've also **checked the pagination code against the documentation** here on GitHub, there's no differences. As I say, the pagination and blog posts were working as intended before the most recent update.

I'll attach some images for visual reference of how my Blog posts and Blog post pages are setup.

Blog posts page (**Note: I am not using the category features on this blog feed**):

![image](https://cloud.githubusercontent.com/assets/23475543/20303315/c8ac5b88-ab22-11e6-9fb4-8d3bcc591200.png)

![image](https://cloud.githubusercontent.com/assets/23475543/20303353/e8bed90a-ab22-11e6-9b82-e13fb4dec2b1.png)

Blog post page:

![image](https://cloud.githubusercontent.com/assets/23475543/20303414/2b2ea356-ab23-11e6-9ad9-b108c8f6878a.png)

![image](https://cloud.githubusercontent.com/assets/23475543/20303426/37c05092-ab23-11e6-8a8e-41572fb71a18.png)

**Comment by CptMeatball** [comment]
What does your front-end code look like? Have you put this in a `partial` for custom styling etc?

**Comment by mjdwebdesign** [comment]
Appreciate the replies.
I thought the same in terms of configuration, even the front-end seems to match the documentation. However, maybe you'll spot something I've missed or simply wouldn't be able to notice.

![image](https://cloud.githubusercontent.com/assets/23475543/20466268/ffc331a2-af67-11e6-8dba-55534fc672f1.png)

Alright so, the screenshot above it the code located within "default.htm" located at "/website/plugins/rainlab/components/posts/deafult.htm".

In terms of the content above the pagination there are some differences however the pagination itself is exactly the same as the current documentation (apart from the arrows inside the previous and next buttons).

Myself personally I can't see anything that is causing the pagination to have issues. Even the changed HTML above the pagination, I can't see that being the source of the problem?

It simply displays the blog's image along with the blog list.

**Comment by CptMeatball** [comment]
Have you edited the component property `Posts per Page` to for example `2` and do you have enough `published` blog posts?

**About editing partials:**
It's a bad practice to directly edit the plugin partial (e.g. `rainlab/blog/components/**/default.htm`) . You should create a custom partial as described here: [Docs: Overriding Partials](https://octobercms.com/docs/cms/components#overriding-partials).

**Comment by mjdwebdesign** [comment]
At the moment I have the "Posts per Page" property set to 2, I have 4 blog posts at current so, I don't think lack of blog posts is the source of the issue. When I change the "Posts per Page" to 4 or more, the pagination does not appear as expected. Then, when I change the value to 3 or lower, the pagination appears again.

I'm pretty new to OctoberCMS and the way it works so, thank you very much for the information about overriding the partials. I'll give this a go when I next have the chance and report back here.

Since we have a small number of blog posts at current, I'll attempt reinstalling the entire plugin and going from there.

**Comment by mjdwebdesign** [comment]
Alright, I completely re-installed the plugin and the pagination now functions as it did before. **Note to self**, don't attempt editing raw source code...

How does overriding partials work then?
My original intention when I changed the source was to edit how the blog posts page and blog post page appeared to the user. Here's what I changed;

- I moved the blog post's featured image above the post's title.
- I added some text underneath the blog post's title to say " Posted by 'user' on 'date' "

From what I can tell reading the documentation personally is, I need to create a partial within the CMS and then assign either the blogPosts or blogPost components to that partial to override the source code, correct?

Do I need to copy and paste the source code into the override partial I create and edit it from there?
